apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -243,6 +242,22 @@ public static Method getIndyBootstrapMethod(final Logger logger) {
         return indyBootstrapDispatcherClass;
     }
 
+    private static Class<?> loadClassInBootstrap(String className, String resourceName) throws IOException, ClassNotFoundException {
+        Class<?> bootstrapClass;
+        try {
+            bootstrapClass = Class.forName(className, false, null);
```

Comment:
No, it's not for tests, and there was a similar construct previously. In this case if the class has already been loaded in the bootstrap CL it allows to avoid trying to inject it from the classpath resource.

Given the original class name is in the `bootstrap.*` package it can't be loaded even if the original `.class` file is in the classpath. I'll add a comment to clarify that.